File: 0075-sort-colors/0075-sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let low = 0, mid = 0, high = nums.length -1 ;
    while(mid <= high) {
        switch(nums[mid]){
            case 0:
            swap(nums, low, mid);
            low++;
            mid++;
            break;

            case 1:
            mid++;
            break;

            case 2:
            swap(nums, mid, high);
            high--;
            break;
        }
    }
};
function swap(arr, index1, index2){
    let temp = arr[index1];
    arr[index1] = arr[index2];
    arr[index2] = temp;
}

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int i=-1, j=nums.length, k=0;
        while(k<j) {
            switch (nums[k]) {
                case 0:
                    i++;
                    swap(i, k, nums);
                    k++;
                break;
                    
                case 1:
                    k++;
                break;

                case 2:
                    j--;
                    swap(j, k, nums);
                break;
            }
        }
    }

    public void swap(int i, int j, int[] nums) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}